ddpui/core/dbt_automation/json_keys_sql.py

```
@@ -0,0 +1,159 @@
+            WITH sample AS (
+                SELECT {quoted_column} AS payload
+                FROM {bq_table}
+                WHERE {quoted_column} IS NOT NULL
+                LIMIT {sample_limit}
+            )
+            SELECT DISTINCT key
+            FROM sample
+            CROSS JOIN UNNEST(
+                IFNULL(JSON_KEYS(SAFE.PARSE_JSON(CAST(sample.payload AS STRING))), [])
+            ) AS key
+        """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)